v2/workloadapi/client.go

```diff
@@ -364,28 +370,38 @@ func (c *Client) watchJWTBundles(ctx context.Context, watcher JWTBundleWatcher)
 	}
 
 	for {
+		c.opts.log.Debugf("Watching JWT bundles")
 		resp, err := stream.Recv()
 		if err != nil {
 			return err
 		}
 
 		bundles := []*jwtbundle.Bundle{}
+		finish := true
```

Comment:
same advice here regarding creating a function to parse the response to simplify this for loop